huggingface-gpt2/ort_addon/ort_supplement/src/transformers/trainer_ort.py

```diff
@@ -252,10 +147,48 @@ def train(self, model_path: Optional[str] = None):
             t_total = int(len(train_dataloader) // self.args.gradient_accumulation_steps * self.args.num_train_epochs)
             num_train_epochs = self.args.num_train_epochs
 
-        scheduler = linear_schedule_with_warmup(num_warmup_steps=self.args.warmup_steps, num_training_steps=t_total)
+        config = self.model.config
+        model_desc = self.gpt2_model_description(config.n_head, 
+                                                    config.vocab_size, 
+                                                    config.n_embd, 
+                                                    config.n_layer, 
+                                                    config.n_ctx, 
+                                                    self.args.per_gpu_train_batch_size)
 
-        loss_scaler = LossScaler(self.ort_model.loss_scale_input_name, True, up_scale_window=2000, loss_scale=float(1 << 20)) if self.args.fp16 else 1
+        from onnxruntime.capi._pybind_state import set_arena_extend_strategy, ArenaExtendStrategy
```

Comment:
Can we do this without using the old API?

Reply:
I checked, and we cannot as of rc3.